transport_nantes/press/views.py

```diff
@@ -165,3 +144,74 @@ class PressMentionDetailView(PermissionRequiredMixin, DetailView):
     model = PressMention
     template_name = "press/press_detail.html"
     permission_required = 'press.press-editor'
+
+
+def fetching_open_graph_data(form=False, presmention_id=None):
+    """
+    This function allow fetch open graph data of an url from
+    a form or from the get data of the url and return
+    a django message.
+    """
+    if form:
+        url = form.cleaned_data['article_link']
+        # Settings the default data
+        form.instance.og_title = form.cleaned_data['article_title']
+        form.instance.og_description = form.cleaned_data['article_summary']
+    else:
+        press_mention = PressMention.objects.get(
+            pk=presmention_id)
+        url = press_mention.article_link
+    try:
+        page = requests.get(url)
+    except Exception as e:
+        logger.warning(f"Can't acess to the website {e}.")
+        return None
+    tree = html.fromstring(page.content.decode("utf-8"))
+    og_title = tree.xpath('//meta[@property="og:title"]/@content')
+    og_description = tree.xpath(
+        '//meta[@property="og:description"]/@content')
+    og_image = tree.xpath('//meta[@property="og:image"]/@content')
+    if not og_title or not og_description:
+        logger.warning("This website doesn't have all open graph data.")
+        return None
+    else:
+        if form:
+            form.instance.og_title = og_title[0]
+            form.instance.og_description = og_description[0]
+        else:
+            press_mention.og_title = og_title[0]
+            press_mention.og_description = og_description[0]
+    try:
+        resp = requests.get(og_image[0])
+    except IndexError:
+        logger.warning(f"This website doesn't have open graph image.")
```

Comment:
f-string is missing placeholders flake8(F541)

Reply:
Fixed